pyproject.toml

```diff
@@ -96,9 +96,42 @@ bson=["py.typed", "*.pyi"]
 pymongo=["py.typed", "*.pyi"]
 gridfs=["py.typed", "*.pyi"]
 
+[tool.pytest.ini_options]
+minversion = "7"
+addopts = ["-ra", "--strict-config", "--strict-markers", "--junitxml=xunit-results/TEST-results.xml"]
+testpaths = ["test"]
+log_cli_level = "INFO"
+norecursedirs = ["test/*"]
+faulthandler_timeout = 1500
+xfail_strict = true
+filterwarnings = [
+    "error",
+    "module:use an explicit session with no_cursor_timeout=True:UserWarning",
+    "module:serverselectiontimeoutms must be:UserWarning",
+    "module:Unsupported compressor:UserWarning",
+    "module:zlibcompressionlevel must be:UserWarning",
+    "module:Wire protocol compression with:UserWarning",
+    # TODO: Remove as part of PYTHON-3923.
+    "module:unclosed <socket.socket:ResourceWarning",
+    "module:unclosed <ssl.SSLSocket:ResourceWarning",
+]
+
+[tool.mypy]
+python_version = "3.7"
+strict = true
```

Comment:
Is strict mode supported in pymongo+gridfs now?

Reply:
Not entirely, we're explicitly disabling `["type-arg", "no-any-return"]`.